tests/src/buildApplication/middlewareManifest.test.ts

```diff
@@ -63,7 +61,7 @@ describe('parseMiddlewareManifest', () => {
 					matchers: [{ regexp: 'regexpF' }],
 				},
 			},
-		};
+		} as unknown as MiddlewareManifest;
```

Comment:
Out of curiosity, why the need for the double cast here?
Is this object not a valid `MiddlewareManifest` by itself?

Reply:
We need it because we don't provide all the fields of the functions:
![Screenshot 2023-03-24 at 09 52 45](https://user-images.githubusercontent.com/61631103/227490323-88f05f21-4c40-4b35-abae-83b641ce81f5.png)
So typescript doesn't seem to accept a single cast 😕

Alternatively if I did a `DeepPartial` we could reduce the double cast into a simple one:
![Screenshot 2023-03-24 at 09 58 01](https://user-images.githubusercontent.com/61631103/227490900-57288a39-ba26-4ae4-ad91-76c3818561fe.png)

Or I could provide all the values, probably with utility otherwise things would get quite verbose and not having any casting, something like:
![Screenshot 2023-03-24 at 10 08 37](https://user-images.githubusercontent.com/61631103/227492621-d3523dca-4008-459d-9009-c5e27b9f1185.png)

Do you have any preference? (or a different option?)

Reply:
Do we really care about the accuracy of the `MiddlewareManifest` interface, since all we do is convert such an object to hydrated middleware? Could we make this stuff partial throughout?

But anyway, I don't mind either way, was just interested. Not a blocker from merging.

Reply:
mh... I made fields in the MiddlewareManifest type partial 👍

Not sure if that's the correct way, but either way all this MiddlewareManifest related code should hopefully be heavily refactored at some point soon (since it relies on the `.next` directory which we shouldn't rely on), so I don't really mind